cli: server fails to shutdown upon Ctrl+C within 30 seconds

The following tests appear to have failed:

[#235178](https://teamcity.cockroachdb.com/viewLog.html?buildId=235178):

```
--- FAIL: acceptance/TestDockerCLI (156.570s)
test_log_scope.go:79: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestDockerCLI472476126
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestDockerCLI/test_server_sig.tcl (35.530s)
cli_test.go:73: non-zero exit code: 1

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/cli/interactive_tests/test_log_flags.tcl
+++ b/pkg/cli/interactive_tests/test_log_flags.tcl
@@ -15,8 +15,7 @@ eexpect ":/# "
 # Check that log files are created by default in the store directory.
 send "$argv start --insecure --store=path=mystore\r"
 eexpect "node starting"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
 send "ls mystore/logs\r"
 eexpect "cockroach.log"
@@ -25,8 +24,7 @@ eexpect ":/# "
 # Check that an empty `-log-dir` disables file logging.
 send "$argv start --insecure --store=path=mystore2 --log-dir=\r"
 eexpect "node starting"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
 send "ls mystore2/logs 2>/dev/null | wc -l\r"
 eexpect "0"
@@ -40,29 +38,22 @@ eexpect ":/# "
 # Check that the user can override.
 send "$argv start --insecure --log-dir=blah/\~/blah\r"
 eexpect "logs: *blah/~/blah"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
 
 # Check that TRUE and FALSE are valid values for the severity flags.
 send "$argv start --insecure --logtostderr=false\r"
 eexpect "node starting"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
-
 send "$argv start --insecure --logtostderr=true\r"
 eexpect "node starting"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
-
 send "$argv start --insecure --logtostderr=2\r"
 eexpect "node starting"
-interrupt
-interrupt
+send "\003"
 eexpect ":/# "
-
 send "$argv start --insecure --logtostderr=cantparse\r"
 eexpect "parsing \"cantparse\": invalid syntax"
 eexpect ":/# "

--- a/pkg/cli/interactive_tests/test_missing_log_output.tcl
+++ b/pkg/cli/interactive_tests/test_missing_log_output.tcl
@@ -2,9 +2,6 @@
 
 source [file join [file dirname $argv0] common.tcl]
 
-# flaky: #15391
-exit 0
-
 spawn /bin/bash
 send "PS1=':''/# '\r"
 eexpect ":/# "

--- a/pkg/cli/interactive_tests/test_server_restart.tcl
+++ b/pkg/cli/interactive_tests/test_server_restart.tcl
@@ -2,9 +2,6 @@
 #
 source [file join [file dirname $argv0] common.tcl]
 
-# flaky: #15391
-exit 0
-
 start_server $argv
 
 spawn /bin/bash

--- a/pkg/storage/replica.go
+++ b/pkg/storage/replica.go
@@ -2160,15 +2160,19 @@ func (r *Replica) tryExecuteWriteBatch(
 		return nil, roachpb.NewError(err), proposalNoRetry
 	}
 
-	// Add the write to the command queue to gate subsequent overlapping
-	// commands until this command completes. Note that this must be
-	// done before getting the max timestamp for the key(s), as
-	// timestamp cache is only updated after preceding commands have
-	// been run to successful completion.
-	log.Event(ctx, "command queue")
-	endCmds, err := r.beginCmds(ctx, &ba, spans)
-	if err != nil {
-		return nil, roachpb.NewError(err), proposalNoRetry
+	var endCmds *endCmds
+	if !ba.IsLeaseRequest() {
+		// Add the write to the command queue to gate subsequent overlapping
+		// commands until this command completes. Note that this must be
+		// done before getting the max timestamp for the key(s), as
+		// timestamp cache is only updated after preceding commands have
+		// been run to successful completion.
+		log.Event(ctx, "command queue")
+		var err error
+		endCmds, err = r.beginCmds(ctx, &ba, spans)
+		if err != nil {
+			return nil, roachpb.NewError(err), proposalNoRetry
+		}
 	}
 
 	// Guarantee we remove the commands from the command queue. This is

